src/ucp/wireup/wireup.c

```diff
@@ -1803,5 +1803,44 @@ unsigned ucp_ep_init_flags(const ucp_worker_h worker,
     return flags;
 }
 
+double ucp_wireup_iface_lat_distance_v1(ucp_context_h context,
+                                        const ucs_linear_func_t *latency,
+                                        const ucs_sys_dev_distance_t *distance)
+{
+    return (context->config.ext.proto_enable) ?
+        (latency->c + distance->latency) : latency->c;
+}
+
+double ucp_wireup_iface_lat_distance_v2(ucp_context_h context,
+                                        const ucs_linear_func_t *latency,
+                                        const ucs_sys_dev_distance_t *distance)
+{
+    ucs_linear_func_t lat;
+
+    if (context->config.ext.proto_enable)
+    {
+        lat    = *latency;
+        lat.c += distance->latency;
+        return ucp_tl_iface_latency(context, &lat);
+    } else {
+        return ucp_tl_iface_latency(context, latency);
+    }
+}
+
+double ucp_wireup_iface_bw_distance(ucp_context_h context,
+                                    const uct_ppn_bandwidth_t *bandwidth,
+                                    const ucs_sys_dev_distance_t *distance)
+{
+    uct_ppn_bandwidth_t bw;
+
+    if (context->config.ext.proto_enable) {
+        bw.shared    = ucs_min(bandwidth->shared, distance->bandwidth);
+        bw.dedicated = ucs_min(bandwidth->dedicated, distance->bandwidth);
```

Comment:
how is it different from what is done in `ucp_worker_iface_open` for proto v1? Why need to handle it differently there then?

Reply:
ucp_worker_iface_open overwrites the latency and bandwidth of the local iface. Here we use the distance info to set the latency and bandwidth for the packed address.